Subproject build issue on Tizen7.0

# Issue 
 
- [x] When compiling with **ruy** enablement:
```
[   93s] c++ -Isubprojects/ruy/libruy_system_aligned_alloc.a.p -Isubprojects/ruy -I../subprojects/ruy -Isubprojects/ruy/__CMake_build -I../subprojects/ruy/__CMake_build -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=gnu++14 -O2 -g2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -O2 -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -Wundef -O3 -Wno-error=unused-result -Wno-error=comment -Wno-error=unknown-pragmas -Wno-error=unused-function -Wno-error=maybe-uninitialized -Wno-error=stringop-overread -Wno-error=array-parameter -MD -MQ subprojects/ruy/libruy_system_aligned_alloc.a.p/ruy_system_aligned_alloc.cc.o -MF subprojects/ruy/libruy_system_aligned_alloc.a.p/ruy_system_aligned_alloc.cc.o.d -o subprojects/ruy/libruy_system_aligned_alloc.a.p/ruy_system_aligned_alloc.cc.o -c ../subprojects/ruy/ruy/system_aligned_alloc.cc
[   93s] cc1plus: error: '-Werror=stringop-overread': no option -Wstringop-overread
[   93s] cc1plus: error: '-Werror=array-parameter': no option -Warray-parameter
[   93s] [38/436] Compiling C object subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_arm_linux_chipset.c.o
[   93s] FAILED: subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_arm_linux_chipset.c.o 
[   93s] cc -Isubprojects/ruy/libcpuinfo.a.p -Isubprojects/ruy -I../subprojects/ruy -I../subprojects/ruy/third_party/cpuinfo/src -I../subprojects/ruy/third_party/cpuinfo/include -I../subprojects/ruy/third_party/cpuinfo/deps/clog/include -Isubprojects/ruy/__CMake_build -I../subprojects/ruy/__CMake_build -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c99 -O2 -g2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -O2 -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -DCPUINFO_LOG_LEVEL=2 -D_GNU_SOURCE=1 -Wno-implicit-function-declaration -Wno-error=unused-result -Wno-error=comment -Wno-error=unknown-pragmas -Wno-error=unused-function -Wno-error=maybe-uninitialized -Wno-error=stringop-overread -Wno-error=array-parameter -MD -MQ subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_arm_linux_chipset.c.o -MF subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_arm_linux_chipset.c.o.d -o subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_arm_linux_chipset.c.o -c ../subprojects/ruy/third_party/cpuinfo/src/arm/linux/chipset.c
[   93s] cc1: error: '-Werror=stringop-overread': no option -Wstringop-overread
[   93s] cc1: error: '-Werror=array-parameter': no option -Warray-parameter
[   93s] [39/436] Compiling C object subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_api.c.o
[   93s] FAILED: subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_api.c.o 
[   93s] cc -Isubprojects/ruy/libcpuinfo.a.p -Isubprojects/ruy -I../subprojects/ruy -I../subprojects/ruy/third_party/cpuinfo/src -I../subprojects/ruy/third_party/cpuinfo/include -I../subprojects/ruy/third_party/cpuinfo/deps/clog/include -Isubprojects/ruy/__CMake_build -I../subprojects/ruy/__CMake_build -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c99 -O2 -g2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -O2 -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -DCPUINFO_LOG_LEVEL=2 -D_GNU_SOURCE=1 -Wno-implicit-function-declaration -Wno-error=unused-result -Wno-error=comment -Wno-error=unknown-pragmas -Wno-error=unused-function -Wno-error=maybe-uninitialized -Wno-error=stringop-overread -Wno-error=array-parameter -MD -MQ subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_api.c.o -MF subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_api.c.o.d -o subprojects/ruy/libcpuinfo.a.p/third_party_cpuinfo_src_api.c.o -c ../subprojects/ruy/third_party/cpuinfo/src/api.c
[   93s] cc1: error: '-Werror=stringop-overread': no option -Wstringop-overread
[   93s] cc1: error: '-Werror=array-parameter': no option -Warray-parameter
[   93s] [40/436] Compiling C++ object subprojects/ruy/libruy_wait.a.p/ruy_wait.cc.o
[   93s] FAILED: subprojects/ruy/libruy_wait.a.p/ruy_wait.cc.o 
[   93s] c++ -Isubprojects/ruy/libruy_wait.a.p -Isubprojects/ruy -I../subprojects/ruy -Isubprojects/ruy/__CMake_build -I../subprojects/ruy/__CMake_build -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=gnu++14 -O2 -g2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -O2 -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -Wundef -O3 -Wno-error=unused-result -Wno-error=comment -Wno-error=unknown-pragmas -Wno-error=unused-function -Wno-error=maybe-uninitialized -Wno-error=stringop-overread -Wno-error=array-parameter -MD -MQ subprojects/ruy/libruy_wait.a.p/ruy_wait.cc.o -MF subprojects/ruy/libruy_wait.a.p/ruy_wait.cc.o.d -o subprojects/ruy/libruy_wait.a.p/ruy_wait.cc.o -c ../subprojects/ruy/ruy/wait.cc
[   93s] cc1plus: error: '-Werror=stringop-overread': no option -Wstringop-overread
[   93s] cc1plus: error: '-Werror=array-parameter': no option -Warray-parameter
[   93s] ninja: build stopped: subcommand failed.
[   93s] error: Bad exit status from /var/tmp/rpm-tmp.GMa84W (%build)
[   93s] 
[   93s] 
[   93s] RPM build errors:
[   93s]     Bad exit status from /var/tmp/rpm-tmp.GMa84W (%build)
[   93s] 
[   93s] ejyang-400TEA-400SEA failed "build nntrainer.spec" at Thu Jul 10 02:20:12 UTC 2025.
[   93s] 
```
- [ ] When compiling with **ggml** enablement:
```
[   20s] [4/377] Compiling C object subprojects/ggml/libggml_cpu.so.p/src_ggml-cpu_ggml-cpu-quants.c.o
[   20s] FAILED: subprojects/ggml/libggml_cpu.so.p/src_ggml-cpu_ggml-cpu-quants.c.o 
[   20s] cc -Isubprojects/ggml/libggml_cpu.so.p -Isubprojects/ggml -I../subprojects/ggml -I../subprojects/ggml/src -I../subprojects/ggml/src/ggml-cpu -I../subprojects/ggml/include -Isubprojects/ggml/__CMake_build -I../subprojects/ggml/__CMake_build -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=gnu11 -O2 -g2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -fPIC -O2 -g2 -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong -Wformat-security -fmessage-length=0 -frecord-gcc-switches -Wl,-z,relro,--as-needed -feliminate-unused-debug-types -Wformat -march=armv8-a+fp+simd+crc+crypto -mtune=cortex-a57.cortex-a53 -g -O3 -DNDEBUG -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wcast-qual -Wno-unused-function -Wdouble-promotion -fopenmp -DGGML_BACKEND_BUILD -DGGML_BACKEND_SHARED -DGGML_SCHED_MAX_COPIES=4 -DGGML_SHARED -DGGML_USE_CPU_AARCH64 -DGGML_USE_OPENMP -D_GNU_SOURCE -D_XOPEN_SOURCE=600 -Dggml_cpu_EXPORTS -Wno-error=pointer-to-int-cast -Wno-error=maybe-uninitialized -MD -MQ subprojects/ggml/libggml_cpu.so.p/src_ggml-cpu_ggml-cpu-quants.c.o -MF subprojects/ggml/libggml_cpu.so.p/src_ggml-cpu_ggml-cpu-quants.c.o.d -o subprojects/ggml/libggml_cpu.so.p/src_ggml-cpu_ggml-cpu-quants.c.o -c ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c
[   20s] In file included from ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:7:
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_q3_K_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: implicit declaration of function 'vld1q_s8_x4'; did you mean 'vld1q_s8_x2'? [-Werror=implicit-function-declaration]
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:5763:48: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  5763 |             const ggml_int8x16x4_t q8bytes_1 = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                                                ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: invalid initializer
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:5763:48: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  5763 |             const ggml_int8x16x4_t q8bytes_1 = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                                                ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: invalid initializer
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:5764:48: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  5764 |             const ggml_int8x16x4_t q8bytes_2 = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                                                ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_q5_K_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: invalid initializer
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:7658:46: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  7658 |             const ggml_int8x16x4_t q8bytes = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                                              ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_q6_K_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:296:27: error: implicit declaration of function 'vld1q_u8_x4'; did you mean 'vld1q_u8_x2'? [-Werror=implicit-function-declaration]
[   20s]   296 | #define ggml_vld1q_u8_x4  vld1q_u8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:8614:40: note: in expansion of macro 'ggml_vld1q_u8_x4'
[   20s]  8614 |             ggml_uint8x16x4_t q6bits = ggml_vld1q_u8_x4(q6); q6 += 64;
[   20s]       |                                        ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:296:27: error: invalid initializer
[   20s]   296 | #define ggml_vld1q_u8_x4  vld1q_u8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:8614:40: note: in expansion of macro 'ggml_vld1q_u8_x4'
[   20s]  8614 |             ggml_uint8x16x4_t q6bits = ggml_vld1q_u8_x4(q6); q6 += 64;
[   20s]       |                                        ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: invalid initializer
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:8615:40: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  8615 |             ggml_int8x16x4_t q8bytes = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                                        ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:8640:23: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  8640 |             q8bytes = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                       ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq2_xxs_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:9536:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  9536 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq2_xs_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:9907:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s]  9907 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq2_s_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:10493:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 10493 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq3_xxs_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:10959:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 10959 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq3_s_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:11309:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 11309 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq1_s_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:11832:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 11832 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq1_m_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:12189:19: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 12189 |             q8b = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                   ^~~~~~~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c: In function 'ggml_vec_dot_iq4_xs_q8_K':
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-impl.h:298:27: error: incompatible types when assigning to type 'int8x16x4_t' {aka 'struct int8x16x4_t'} from type 'int'
[   20s]   298 | #define ggml_vld1q_s8_x4  vld1q_s8_x4
[   20s]       |                           ^~~~~~~~~~~
[   20s] ../subprojects/ggml/src/ggml-cpu/ggml-cpu-quants.c:12698:22: note: in expansion of macro 'ggml_vld1q_s8_x4'
[   20s] 12698 |             q8b    = ggml_vld1q_s8_x4(q8); q8 += 64;
[   20s]       |                      ^~~~~~~~~~~~~~~~
[   20s] cc1: all warnings being treated as errors
```

# Comment
- In draft #3311, options are added to enable submodule compilation for Tizen GBS builds.
- The draft primarily focuses on aligning the enable-ggml feature with the meson_option, so it does not represent a groundbreaking solution.
- This issue should be addressed.

--- a/meson.build
+++ b/meson.build
@@ -600,8 +600,10 @@ else
     ruy_flags += '-Wno-error=uninitialized'
   else
     ruy_flags += '-Wno-error=maybe-uninitialized'
-    ruy_flags += '-Wno-error=stringop-overread'
-    ruy_flags += '-Wno-error=array-parameter'
+    if cxx.has_argument('-Wno-error=stringop-overread') and cxx.has_argument('-Wno-error=array-parameter')
+      ruy_flags += '-Wno-error=stringop-overread'
+      ruy_flags += '-Wno-error=array-parameter'
+    endif
   endif
   
   ruy_options.append_compile_args('c', '-Wno-implicit-function-declaration')

--- a/meson.build
+++ b/meson.build
@@ -575,6 +575,8 @@ else
 endif
 
 # Configure the Ruy project (CMake)
+ruy_dep = dummy_dep
+if get_option('enable-ruy')
 if get_option('platform') == 'android'
   ruy_root = subprojects_dir_absolute / 'ruy'
   ruy_dep = declare_dependency()
@@ -609,6 +611,7 @@ else
   ruy_proj = cmake.subproject('ruy', options: ruy_options, required: true)
   ruy_dep = ruy_proj.dependency('ruy').partial_dependency(includes: true)
 endif
+endif # enable ruy
 
 if get_option('platform') == 'android'
   message('preparing ml api')

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -48,6 +48,7 @@ option('enable-opencl', type: 'boolean', value: false)
 option('enable-biqgemm', type: 'boolean', value: false)
 option('biqgemm-path', type: 'string', value: '../BiQGEMM')
 option('enable-benchmarks', type: 'boolean', value : false)
+option('enable-ruy', type: 'boolean', value: true)
 option('ggml-thread-backend', type: 'string', value: 'mixed')
 
 # ml-api dependency (to enable, install capi-inference from github.com/nnstreamer/api )

--- a/packaging/nntrainer.spec
+++ b/packaging/nntrainer.spec
@@ -1,9 +1,12 @@
-# Execute gbs with --define "testcoverage 1" in case that you must get unittest coverage statistics
+ #Execute gbs with --define "testcoverage 1" in case that you must get unittest coverage statistics
 %define         use_cblas 1
 %define         nnstreamer_filter 1
 %define         nnstreamer_trainer 1
 %define         nnstreamer_subplugin_path lib/nnstreamer
 %define         use_gym 0
+%define         use_ggml 1
+%define         use_ruy 1
+%define         use_biqgemm 0
 %define         support_ccapi 1
 %define         support_nnstreamer_backbone 1
 %define         support_tflite_backbone 1
@@ -327,9 +330,11 @@ Requires:	nnstreamer-nntrainer-trainer = %{version}-%{release}
 NNSteamer tensor trainer static package for nntrainer to support inference.
 %endif #nnstreamer_trainer
 
+%if 0%{?use_ruy}
 %package -n ruy
 Summary: Ruy support in NNTrainer
 %description -n ruy
+%endif
 
 %if %{with gpu}
 %package -n clblast
@@ -373,6 +378,24 @@ Summary: CLBlast as an OpenCL backend for BLAS operations in NNTrainer
 # Using cblas for Matrix calculation
 %if 0%{?use_cblas}
 %define enable_cblas -Denable-blas=true
+%endif
+
+%if 0%{?use_ggml}
+%define enable_ggml -Denable-ggml=true
+%else
+%define enable_ggml -Denable-ggml=false
+%endif
+
+%if 0%{?use_ruy}
+%define enable_ruy -Denable-ruy=true
+%else
+%define enable_ruy -Denable-ruy=false
+%endif
+
+%if 0%{?use_biqgemm}
+%define enable_biqgemm -Denable-biqgemm=true
+%else
+%define enable_biqgemm -Denable-biqgemm=false
 %endif
 
 %if 0%{?support_nnstreamer_backbone}
@@ -424,7 +447,9 @@ ln -sf %{_libdir}/pkgconfig/capi-nnstreamer.pc %{_libdir}/pkgconfig/capi-ml-comm
 %endif
 
 # Setup Ruy
+%if 0%{?use_ruy}
 tar -xf packaging/ruy.tar.gz -C subprojects
+%endif
 
 # Setup CLBlast
 %if %{with gpu}
@@ -559,15 +584,20 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %{_includedir}/nntrainer/memory_data.h
 %{_includedir}/nntrainer/tensor.h
 %{_includedir}/nntrainer/tensor_base.h
+%if 0%{?use_ggml}
 %{_includedir}/nntrainer/q4_k_tensor.h
 %{_includedir}/nntrainer/q6_k_tensor.h
 %{_includedir}/nntrainer/q4_0_tensor.h
+%endif
 %{_includedir}/nntrainer/int4_tensor.h
 %{_includedir}/nntrainer/uint4_tensor.h
 %{_includedir}/nntrainer/char_tensor.h
 %{_includedir}/nntrainer/short_tensor.h
 %{_includedir}/nntrainer/uint_tensor.h
 %{_includedir}/nntrainer/float_tensor.h
+%if 0%{?use_biqgemm}
+%{_includedir}/nntrainer/bcq_tensor.h
+%endif
 %if 0%{?enable_fp16}
 %{_includedir}/nntrainer/half_tensor.h
 %endif
@@ -644,9 +674,9 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 # update this to enable external applications
 # @todo filter out headers that should be hidden, and classifiy in the appropriate place if not
 %{_includedir}/nntrainer/util_func.h
-/usr/include/nntrainer/noncopyable.h
-/usr/include/nntrainer/nonmovable.h
-/usr/include/nntrainer/singleton.h
+%{_includedir}/nntrainer/noncopyable.h
+%{_includedir}/nntrainer/nonmovable.h
+%{_includedir}/nntrainer/singleton.h
 %{_includedir}/nntrainer/fp16.h
 %{_includedir}/nntrainer/util_simd.h
 %{_includedir}/nntrainer/dynamic_library_loader.h
@@ -764,6 +794,7 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %endif #nnstreamer_trainer
 
 # Ruy
+%if 0%{?use_ruy}
 %files -n ruy
 %manifest nntrainer.manifest
 %defattr(-,root,root,-)
@@ -777,6 +808,7 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %ifarch x86_64
 %{_bindir}/cpuid_dump
 %endif #x86_64
+%endif
 
 # CLBlast
 %if %{with gpu}
@@ -808,7 +840,7 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %defattr(-,root,root,-)
 %license LICENSE
 %{_libdir}/nntrainer/bin/applications/*
-%{_libdir}/nntrainer/layers/*layer.so
+%{_libdir}/nntrainer/layers/*layer.so 
 
 %if 0%{?gcov:1}
 %files gcov

--- a/packaging/nntrainer.spec
+++ b/packaging/nntrainer.spec
@@ -4,7 +4,6 @@
 %define         nnstreamer_trainer 1
 %define         nnstreamer_subplugin_path lib/nnstreamer
 %define         use_gym 0
-%define         use_ggml 1
 %define         use_ruy 1
 %define         use_biqgemm 0
 %define         support_ccapi 1
@@ -380,11 +379,6 @@ Summary: CLBlast as an OpenCL backend for BLAS operations in NNTrainer
 %define enable_cblas -Denable-blas=true
 %endif
 
-%if 0%{?use_ggml}
-%define enable_ggml -Denable-ggml=true
-%else
-%define enable_ggml -Denable-ggml=false
-%endif
 
 %if 0%{?use_ruy}
 %define enable_ruy -Denable-ruy=true
@@ -584,11 +578,9 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %{_includedir}/nntrainer/memory_data.h
 %{_includedir}/nntrainer/tensor.h
 %{_includedir}/nntrainer/tensor_base.h
-%if 0%{?use_ggml}
 %{_includedir}/nntrainer/q4_k_tensor.h
 %{_includedir}/nntrainer/q6_k_tensor.h
 %{_includedir}/nntrainer/q4_0_tensor.h
-%endif
 %{_includedir}/nntrainer/int4_tensor.h
 %{_includedir}/nntrainer/uint4_tensor.h
 %{_includedir}/nntrainer/char_tensor.h

--- a/packaging/nntrainer.spec
+++ b/packaging/nntrainer.spec
@@ -1,4 +1,4 @@
- #Execute gbs with --define "testcoverage 1" in case that you must get unittest coverage statistics
+# Execute gbs with --define "testcoverage 1" in case that you must get unittest coverage statistics
 %define         use_cblas 1
 %define         nnstreamer_filter 1
 %define         nnstreamer_trainer 1
@@ -840,7 +840,7 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
 %defattr(-,root,root,-)
 %license LICENSE
 %{_libdir}/nntrainer/bin/applications/*
-%{_libdir}/nntrainer/layers/*layer.so 
+%{_libdir}/nntrainer/layers/*layer.so
 
 %if 0%{?gcov:1}
 %files gcov

